components/brave_wallet_ui/common/async/handlers.ts

```diff
@@ -400,25 +396,13 @@ handler.on(WalletActions.selectPortfolioTimeline.type, async (store: Store, payl
 handler.on(WalletActions.sendTransaction.type, async (
   store: Store,
   payload:
-    | Omit<SendEthTransactionParams, 'hasEIP1559Support'>
+    | SendEthTransactionParams
     | SendFilTransactionParams
     | SendSolTransactionParams
 ) => {
-  const { wallet: walletState } = store.getState()
-  const selectedNetwork = await getSelectedNetwork(getAPIProxy())
-
   let addResult
   if (payload.coin === BraveWallet.CoinType.ETH) {
-    addResult = await sendEthTransaction({
-      ...payload,
-      hasEIP1559Support:
-        !!selectedNetwork &&
-        !!walletState.selectedAccount &&
-        hasEIP1559Support(
-          walletState.selectedAccount.accountType,
-          selectedNetwork
-        )
-    })
```

Comment:
SendEthTransactionParams now has account and selectedNetwork so hasEIP1559Support will be evaluated deeper when it's actually needed.